### PR TITLE
Tests and usblib changes

### LINKED ISSUE #111: Add tests
## Unit tests 
- [x] AddressGen
- [x] ApplySettings
- [x] BackUp
- [x] Cancel
- [x] CheckMessageSignature
- [x] ChangePin
- [x] Connected
- [x] Available
- [x] FirmwareUpload
- [x] GetFeatures
- [x] GenerateMnemonic
- [x] Recovery
- [x] SetMnemonic
- [x] TransactionSign
- [x] Wipe

## Integration tests
- [x] AddressGen
- [x] ApplySettings
- [x] BackUp
- [x] Cancel
- [x] CheckMessageSignature
- [x] ChangePin
- [x] Connected
- [x] Available
- [x] GetFeatures
- [x] GenerateMnemonic
- [x] Recovery
- [x] SetMnemonic
- [x] TransactionSign
- [x] Wipe

### PR DESCRIPTION
**Pull request by therealssj**
Fixes #111 
Fixes #https://github.com/skycoin/hardware-wallet-daemon/issues/44

Changes:
- Add cli integration tests
- Add more unit tests
- Update usblib

Does this change need to mentioned in CHANGELOG.md?
yes

Requires changes in protobuff specs?
no

Requires testing
yes

change is a test

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,10 +22,13 @@ mocks: ## Create all mock files for unit tests
 test_unit: ## Run unit tests
 	go test -v github.com/skycoin/hardware-wallet-go/src/device-wallet
 
-test_integration: ## Run integration tests
-	go test -count=1 -v github.com/skycoin/hardware-wallet-go/src/device-wallet/integration
+integration-test-emulator: ## Run emulator integration tests
```
Comment (stdevAlDen) [inline Makefile]: @therealssj Just a detail, I personally prefer `test-integration-emulator`, what is grammatically correct is what you did in your changes but I like hit `test` `TAB` `TAB` in the terminal and see all the `test` related rules, this is why I preffer `test-unit` over "Unit tests" and [`release-sign` over "Sigen release"](https://github.com/skycoin/hardware-wallet/blob/160614a3ec458da4e3ca238ecec9f344f54416f9/Makefile#L123-L158)

File: Makefile [hunk 2]
```diff
@@ -20,10 +22,13 @@ mocks: ## Create all mock files for unit tests
...
+	./ci-scripts/integration-test.sh -a -m EMULATOR -n emulator-integration
+
+integration-test-wallet: ## Run usb integration tests
```
Comment (stdevAlDen) [inline Makefile]: @therealssj  https://github.com/skycoin/hardware-wallet-go/pull/116/files#r281313885

**Comment by therealssj** [comment]
@stdevAlDen im making some changes to hwgo right now so this PR is on hold. Will get back to it today

**Comment by stdevAlDen** [comment]
@therealssj 👍

**Pull request closed without merging**